kit/ForKit.cpp

```diff
@@ -383,18 +380,28 @@ static int createLibreOfficeKit(const std::string& childRoot,
     LOG_DBG("Forking a coolkit process with jailId: " << jailId << " as spare coolkit #"
                                                       << spareKitId << '.');
     const auto startForkingTime = std::chrono::steady_clock::now();
-
-    const pid_t pid = fork();
-    if (!pid)
+    pid_t pid = 0;
+    if (Util::isKitInProcess() && !lokitProcessIsRunning)
     {
-        // Child
-
-        // Close the pipe from coolwsd
-        close(0);
+        std::thread ([childRoot, jailId, sysTemplate, loTemplate, queryVersion]
+            {
+                Util::setThreadName("kit_in_process");
+                lokit_main(childRoot, jailId, sysTemplate, loTemplate, NoCapsForKit, NoSeccomp,
+                   queryVersion, DisplayVersion, spareKitId);
+            }).detach();
```

Comment:
Why are we detaching? Why isn't the thread owned by coolwsd?

Can you share the design of this whole PR please?